### ISSUE TITLE
MiniBatch Support in Shifu [Investigation]

### ISSUE THREAD

**Issue by zhangpengshan** [ShifuML/shifu#434]
Separate one global parameter server from master and update gradients in a setting minibatch.

This issue is about the solution, not the implementation.

**Comment by zhangpengshan** [comment]
In each epoch so far Shifu will do batch gradient descent, to support mini batch which can be used to improve the convergence.

Set a parameter like miniBatchRate.

**Comment by zhangpengshan** [comment]
It is doable to use a miniBatchRate, say if miniBatchRate is 0.1， set each iteration to run only on 0.1 of whole data set, the logic is straightfoward in each worker. Previous one iteration, now 10 iterations is needed for one epoch. 

Set it as high priority as Minibatch GD may improve final convergency speed

**Comment by zhangpengshan** [comment]
Done with miniBatchs if config in ModelConfig#train#params
